Automattic/wp-calypso
# Gutenboarding: Calypso-Free Launch Button Injection

## Changes proposed in this Pull Request

**TLDR Version:** A majority of this PR's changes is related to the overhaul of the Launch button injection where most logic to determine which launch flow to load is now moved over to the server-side.

1. **Removed `editor-gutenboarding-launch`. Launch button moved to `editor-site-launch`.** 
Previously this was responsible for injecting launch button on to the top bar. This is now moved over to `editor-site-launch` under `editor-site-launch/src/launch-button/index.ts` to streamline things.

2. **Three (3) separate compilation entry points for `editor-site-launch`.**  
Previously both step-by-step and focused launch are bundled into a single build, now they are splitted out to ensure only the necessary script is loaded (always nice to shave off on bundle size). The `package.json` file was modified to build the following: 
    * `launch-button.ts` => `dist/launch-button.js`
    * `step-by-step-launch.ts` => `dist/step-by-step-launch.js`
    * `focused-launch.ts` => `dist/focused-launch.js`

3. **Server-side determines whether to load Step-By-Step, Focused or Redirect launch flow.**  
This can be seen at D55512-code. When calling `apply_filters('a8c_enable_editor_site_launch')` on the server side, it will return an object containing `site_slug`, `launch_url` and `launch_flow: "gutenboarding-launch" | "focused-launch" | "launch-site"`.

4. **Launch button no longer rely on `window.calypsoifyGutenberg`. It relies by `window.wpcomEditorSiteLaunch` object injected by server-side `editor-site-launch/index.php` when enqueuing scripts.**
This means launch button / launch flow can run without Calypso's iframe. The `window.wpcomEditorSiteLaunch` contains the same data as mentioned in point 3.


**Other considerations:**
1. ~~This would render the client-side `create/focused-launch-flow` feature flag useless but I haven't look into how to hide behind feature flag on server-side.~~ We currently still supporting `create/focused-launch-flow` feature flag but will soon transition to using Explat and variation assignment for testing.
2. There are other places (outside of our launch codes) still relying on data from `calypsoifyGutenberg`, especially the `calypsoifyGutenberg?.isGutenboarding` flag. So any `calypsoifyGutenberg` related cleanup to be pushed to another PR here https://github.com/Automattic/wp-calypso/pull/49067.


## Testing instructions

**Setting up**
1. Apply patch D55512-code.
2. Run `yarn dev --sync` on `apps\editing-toolkit`.

**Test Step-By-Step Launch**
1. Create a **Gutenboarding site** via `/new`.
2. Sandbox the newly created site.
3. Go to the block editor and click on the **Launch** button.
4. It should show **Step-By-Step Launch**.

**Test Focused Launch**
1. Create a **Control site** via `/start`.
2. Sandbox the newly created site.
3. Go to the `focused_launch_flow` experiment in ExPlat/Abacus and click `treatment` to assign this A/B variation to your WP user.
4. Go to the block editor and click on the **Launch** button.
5. It should show the **Focused Launch**.

**Test Redirect Launch**
1. You can use the same Control site above.
2. Go to the `focused_launch_flow` experiment in ExPlat/Abacus and click `control` to assign this A/B variation to your WP user.
3. Go to the block editor and click on the **Launch** button.
4. It should show **redirect to `/start/launch-site`**.



Fixes #46839 #47803
